tests/test_physics.py

```diff
@@ -189,5 +195,96 @@ def test_dynamics(sim):
+    sim.set_gravity(np.array([0, 0, 0.0]))
+
+    sphere_path = osp.abspath("data/test_assets/objects/sphere")
+    template_ids = sim.load_object_configs(sphere_path)
+    object_template = sim.get_object_template(template_ids[0])
+    object_template.set_linear_damping(0.0)
+    object_template.set_angular_damping(0.0)
+
+    for iteration in range(2):
+        sim.reset()
+        object_id = sim.add_object(template_ids[0])
+        vel_control = sim.get_object_velocity_control(object_id)
+
+        if iteration == 0:
+            if (
+                sim.get_object_motion_type(object_id)
+                != habitat_sim.physics.MotionType.DYNAMIC
+            ):
+                # Non-dynamic simulator in use. Skip 1st pass.
+                sim.remove_object(object_id)
+                continue
+        elif iteration == 1:
+            # test KINEMATIC
+            sim.set_object_motion_type(
+                habitat_sim.physics.MotionType.KINEMATIC, object_id
+            )
+
+        # test global velocities
+        vel_control.linear_velocity = np.array([1.0, 0, 0])
+        vel_control.angular_velocity = np.array([0, 1.0, 0])
+        vel_control.controlling_lin_vel = True
+        vel_control.controlling_ang_vel = True
+
+        while sim.get_world_time() < 1.0:
+            # NOTE: stepping close to default timestep to get near-constant velocity control of DYNAMIC bodies.
+            sim.step_physics(0.00416)
+
+        ground_truth_pos = sim.get_world_time() * vel_control.linear_velocity
+        assert np.allclose(sim.get_translation(object_id), ground_truth_pos, atol=0.01)
+        ground_truth_q = mn.Quaternion([[0, 0.480551, 0], 0.876967])
+        angle_error = mn.math.angle(ground_truth_q, sim.get_rotation(object_id))
+        if not math.isnan(angle_error):
+            # nan indicates very low angle error
+            assert angle_error < mn.Rad(0.005)
+
+        sim.reset()
+
+        # test local velocities (turn in a half circle)
+        vel_control.lin_vel_is_local = True
+        vel_control.ang_vel_is_local = True
+        vel_control.linear_velocity = np.array([0, 0, -math.pi])
+        vel_control.angular_velocity = np.array([math.pi * 2.0, 0, 0])
+
+        sim.set_translation(np.array([0, 0, 0.0]), object_id)
+        sim.set_rotation(mn.Quaternion(), object_id)
+
+        while sim.get_world_time() < 0.5:
+            # NOTE: stepping close to default timestep to get near-constant velocity control of DYNAMIC bodies.
+            sim.step_physics(0.00416)
+
+        # NOTE: explicit integration, so expect some error
+        ground_truth_q = mn.Quaternion([[1.0, 0, 0], 0])
+        assert np.allclose(
+            sim.get_translation(object_id), np.array([0, 1.0, 0.0]), atol=0.03
+        )
+        angle_error = mn.math.angle(ground_truth_q, sim.get_rotation(object_id))
+        if not math.isnan(angle_error):
+            # nan indicates very low angle error
+            assert angle_error < mn.Rad(0.05)
```

Comment:
You mentioned this on Slack -- I'll try to fix today so you can remove this piece of code before it's merged in.

Reply:
Fixed in mosra/magnum@498f4c59e1186a46cd8b30547d9383a68498a8a5 -- feel free to update the submodule here.